packages/fast-layouts-react/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -0,0 +1,311 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import manageJss, { ComponentStyles, IJSSManagerProps, IManagedClasses } from "@microsoft/fast-jss-manager-react";
+import Foundation, { IFoundationProps } from "../foundation";
+import { HorizontalOverflowProps } from "./horizontal-overflow.props";
+import { getItems, scrollLeft } from "./horizontal-overflow.utilities";
+
+export enum ButtonDirection {
+    previous = "previous",
+    next = "next"
+}
+
+export interface IHorizontalOverflowClassNamesContract {
+    horizontalOverflow: string;
+    horizontalOverflow_items: string;
+    horizontalOverflow_next: string;
+    horizontalOverflow_previous: string;
+}
+
+const styles: ComponentStyles<IHorizontalOverflowClassNamesContract, {}> = {
+    horizontalOverflow: {
+        position: "relative",
+        overflow: "hidden"
+    },
+    horizontalOverflow_items: {
+        position: "relative",
+        whiteSpace: "nowrap",
+        overflowX: "scroll",
+        "&::-webkit-scrollbar": {
+            display: "none"
+        }
+    },
+    horizontalOverflow_next: {
+        position: "absolute",
+        zIndex: "1",
+        right: "0",
+        top: "0",
+        bottom: "0",
+        "& > div": {
+            display: "flex",
+            alignItems: "center",
+            height: "100%"
+        }
+    },
+    horizontalOverflow_previous: {
+        position: "absolute",
+        zIndex: "1",
+        left: "0",
+        top: "0",
+        bottom: "0",
+        "& > div": {
+            display: "flex",
+            alignItems: "center",
+            height: "100%"
+        }
+    }
+};
+
+export interface IHorizontalOverflowState {
+    items: HTMLElement[];
+    itemsHeight: string;
+}
+
+class HorizontalOverflow extends Foundation<HorizontalOverflowProps, IHorizontalOverflowState> {
+
+    public static getDerivedStateFromProps(
+        nextProps: HorizontalOverflowProps,
+        prevState: IHorizontalOverflowState
+    ): null | Partial<IHorizontalOverflowState> {
```

Comment:
I don't think you need to store items in state. What is the benefit of doing that over excluding next/prev buttons in the the render cycle itself?